backend/box/box.go

```diff
@@ -428,6 +429,7 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 		srv:         rest.NewClient(client).SetRoot(rootURL),
 		pacer:       fs.NewPacer(ctx, pacer.NewDefault(pacer.MinSleep(minSleep), pacer.MaxSleep(maxSleep), pacer.DecayConstant(decayConstant))),
 		uploadToken: pacer.NewTokenDispenser(ci.Transfers),
+		fileIdPaths: new(sync.Map),
```

Comment:
Agreed, but not sure what the alternative is, since the box API doesn't provide the required information.

What I've done in the latest diff is make the map conditional on `--poll-interval` being greater than zero. So if someone has a lot of files and would rather save on memory, they can disable polling.

In the case where they disable/enable polling at runtime with `rclone rc vfs/poll-interval`, I log a warning message that a manual `vfs/refresh` will likely be required.

Reply:
After giving it some further thought, I think saving the `fileIdPaths` map is a bad idea as-is. For example, say a directory with many descendents is renamed. All of its descendents should be removed from the map so that we don't notify for them any more. That's difficult to do if we're just saving full paths as strings.

In the latest diff, I now have a map saving IDs to a tuple of metadata: (ParentID, Name, SequenceID). The first two elements allow the path to be dynamically constructed based on the current `dirCache`.

The third element is unrelated, I noticed it while combing through the Box API in that it allows for ignoring old Events.

Reply:
OK - I've reverted the bits of code that only conditionally populated the map if polling is enabled.